dev/src/reference.ts

```diff
@@ -957,10 +996,14 @@ export class QueryOptions {
    * Returns the union of the current and the provided options.
    * @private
    */
-  with(settings: Partial<QueryOptions>): QueryOptions {
+  with(
+    settings: Partial<QueryOptions<T>>,
+    converter = defaultConverter as FirestoreDataConverter<T>
+  ): QueryOptions<T> {
     return new QueryOptions(
       coalesce(settings.parentPath, this.parentPath)!,
       coalesce(settings.collectionId, this.collectionId)!,
+      converter,
```

Comment:
Hm. This is scary. This resets the converter implicitly, even when it is passed in. If possible, can you drop the `converter` argument and change `settings` to `Partial<Omit<QueryOptions<T>, 'converter'>>`. See https://www.typescriptlang.org/docs/handbook/utility-types.html#omittk

Reply:
Done.